.github/workflows/tests.yml

```diff
@@ -62,11 +62,11 @@ jobs:
       - name: Run unit tests
         run: make unit-tests
 
-      # disabled as 'pinging codecov' hangs
-      # see https://github.com/codecov/feedback/issues/354
-      # - name: Send coverage to codecov.io
-      #   run: bash <(curl -s https://codecov.io/bash)
-
+      - name: Upload results to Codecov
+        uses: codecov/codecov-action@v4
+        with:
+          files: ./coverage.txt
```

Comment:
I'm still unsure that this is needed, but ok :shrug:

Reply:
I think it's better to be explicit and also, as far as I saw it tries `.xml` and `.cov` and others. Not sure about `.txt` (which it was before in our project - no idea why though)